pylint/checkers/exceptions.py

```diff
@@ -550,13 +560,16 @@ def visit_tryexcept(self, node: nodes.TryExcept) -> None:
                                 "bad-except-order", node=handler.type, args=msg
                             )
                     if (
-                        exception.name in self.linter.config.overgeneral_exceptions
+                        exception.qname() in self.linter.config.overgeneral_exceptions
+                        # not a qualified name, deprecated
+                        or "." not in exception.name
+                        and exception.name in self.linter.config.overgeneral_exceptions
                         and exception.root().name == utils.EXCEPTIONS_MODULE
-                        and not _is_raising(handler.body)
                     ):
-                        self.add_message(
-                            "broad-except", args=exception.name, node=handler.type
-                        )
+                        if not _is_raising(handler.body):
+                            self.add_message(
+                                "broad-except", args=exception.name, node=handler.type
```

Comment:
While we're at it, would you mind adding `confidence=INFERENCE` here? We're updating the test output anyway so this is a good time to fix this as well 😄